src/services/review/ReviewRequestService.ts

```diff
@@ -71,70 +100,163 @@ export default class ReviewRequestService {
     this.reviewMeta = reviewMeta
     this.reviewRequestView = reviewRequestView
     this.pageService = pageService
+    this.configService = configService
   }
 
-  compareDiff = async (
-    sessionData: UserWithSiteSessionData,
+  compareDiff = (
+    userWithSiteSessionData: UserWithSiteSessionData,
     stagingLink: StagingPermalink
-  ): Promise<EditedItemDto[]> => {
+  ) =>
+    ResultAsync.fromPromise(
+      this.apiService.getCommitDiff(userWithSiteSessionData.siteName),
+      // TODO: Write a handler for github errors to our own internal errors
+      () => new NetworkError()
+    )
+      .andThen(({ files, commits }) =>
+        ResultAsync.fromPromise(
+          this.computeShaMappings(commits),
+          () => new DatabaseError()
+        ).map((mappings) => ({ mappings, files }))
+      )
+      .andThen(({ mappings, files }) =>
+        combine(
+          this.compareDiffWithMappings(
+            userWithSiteSessionData,
+            stagingLink,
+            files,
+            mappings
+          ).map((res) =>
+            res
+              // NOTE: Need to type-hint so that we can recover
+              .map<BaseEditedItemDto | WithEditMeta<EditedItemDto>>(_.identity)
+              .orElse((baseItem) => okAsync(baseItem))
+          )
+        )
+      )
+
+  private compareDiffWithMappings = (
+    sessionData: UserWithSiteSessionData,
+    stagingLink: StagingPermalink,
+    files: RawFileChangeInfo[],
+    mappings: Record<
+      string,
+      {
+        author: string
+        unixTime: number
+      }
+    >
+  ): ResultAsync<WithEditMeta<EditedItemDto>, BaseEditedItemDto>[] => {
     // Step 1: Get the site name
     const { siteName } = sessionData
 
-    // Step 2: Get the list of changed files using Github's API
-    // Refer here for details; https://docs.github.com/en/rest/commits/commits#compare-two-commits
-    // Note that we need a triple dot (...) between base and head refs
-    const { files, commits } = await this.apiService.getCommitDiff(siteName)
+    return files.map(({ filename, contents_url: contents }) => {
```

Comment:
why are we using snake_case here

Reply:
github returns as snake_case, renamed to `contents`